### ISSUE TITLE
Slayer plugin does not count all alternatives for slayer tasks such as GWD bosses

### ISSUE THREAD

**Issue by brickinthewall4138** [runelite/runelite#16983]
### Type

Incorrect behavior

### Operating System

None

### Bug description

In `net.runelite.client.plugins.slayer.Task` which defines slayer tasks including alternative NPC names, there are definitions such as this:

`BLUE_DRAGONS("Blue dragons", ItemID.BLUE_DRAGON_MASK, "Baby blue dragon"),`

this by default catches blue dragons, but also counts baby blue dragons as being "on task". However, Vorkath isn't in here. There is also this line

`GREATER_DEMONS("Greater demons", ItemID.GREATER_DEMON_MASK),`

as such as far as I can tell K'ril, who counts as a greater demon, is not considered to be "on task". 

The task kill count **does** still go down as expected but this impacts downstream plugins such as the dps-calculator plugin I occasionally contribute to - I noticed while using it that it didn't consider these alternative bosses to be "on task" and wasn't applying the slayer helm bonuses, however that uses this plugin to get whether an NPC is "on task".

There are many, many others than just these - all the GWD bosses I think don't count, and kree'arra's minions don't.

I'd be willing to put in the PR that adds all these NPCs if it's a hassle, just let me know if I'm not barking up the wrong tree first - for example some of these bosses already have their own named tasks which could be problematic.

### RuneLite version

1.10.10.4
2.4.2

**Comment by hex-agon** [comment]
I don't think this would be useful for the core plugin since it now tracks slayer task progress via varbits, having only the task name seems fine to be and this sounds like something that should be handled by hub plugins themselves.
